x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/X-FilterVector.java.st

```diff
@@ -43,7 +40,7 @@ $endif$
 
     @Override
     public $Type$Block asBlock() {
-        return block;
+        return new $Type$VectorBlock(this);
```

Comment:
This led to circular references between vector and block. Since a vector can be used in multiple blocks (via filtering), it should probably not know about the containing block.